build error in http_get sample

```
  CC      subsys/net/lib/sockets/sockets.o
/projects/zephyr2/subsys/net/lib/sockets/sockets.c: In function ‘zsock_sendto’:
/projects/zephyr2/subsys/net/lib/sockets/sockets.c:277:11: error: ‘err’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
   errno = -err;
           ^~~~
cc1: all warnings being treated as errors
make[7]: *** [/projects/zephyr2/scripts/Makefile.build:183: subsys/net/lib/sockets/sockets.o] Error 1
  CC      subsys/net/lib/sockets/getaddrinfo.o
make[7]: Target '__build' not remade because of errors.
make[6]: *** [/projects/zephyr2/scripts/Makefile.build:357: subsys/net/lib/sockets] Error 2
make[6]: Target '__build' not remade because of errors.
make[5]: *** [/projects/zephyr2/scripts/Makefile.build:357: subsys/net/lib] Error 2
```

This is when building for qemu_x86.
Not sure why CI hasn't caught this.

--- a/subsys/net/lib/sockets/sockets.c
+++ b/subsys/net/lib/sockets/sockets.c
@@ -266,10 +266,8 @@ ssize_t zsock_sendto(int sock, const void *buf, size_t len, int flags,
 	if (sock_type == SOCK_DGRAM) {
 		err = net_context_sendto(send_pkt, dest_addr, addrlen, NULL,
 					 timeout, NULL, NULL);
-	} else if (sock_type == SOCK_STREAM) {
-		err = net_context_send(send_pkt, NULL, timeout, NULL, NULL);
 	} else {
-		__ASSERT(0, "Unknown socket type");
+		err = net_context_send(send_pkt, NULL, timeout, NULL, NULL);
 	}
 
 	if (err < 0) {

